service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -61,17 +62,21 @@ public class SpecimenMapper {
     private final MessageContext messageContext;
     private final ObservationMapper observationMapper;
     private final RandomIdGeneratorService randomIdGeneratorService;
+    private final ConfidentialityService confidentialityService;
 
     public String mapSpecimenToCompoundStatement(Specimen specimen, List<Observation> observations, DiagnosticReport diagnosticReport) {
         String availabilityTimeElement = StatementTimeMappingUtils.prepareAvailabilityTime(diagnosticReport.getIssuedElement());
         String mappedObservations = mapObservationsAssociatedWithSpecimen(specimen, observations);
 
+        final Optional<String> confidentialityCode = confidentialityService.generateConfidentialityCode(specimen);
```

Comment:
I think we can move this line into line 79 and remove confidentialityCode variable similarly as it is done in DiagnosticReportMapper